sdks/java/io/file-based-io-tests/src/test/java/org/apache/beam/sdk/io/text/TextIOIT.java

```diff
@@ -83,25 +90,82 @@ private static String appendTimestamp(String filenamePrefix) {
     return String.format("%s_%s", filenamePrefix, new Date().getTime());
   }
 
-  @Test
-  public void writeThenReadAll() {
-    PCollection<String> testFilenames = pipeline
-        .apply("Generate sequence", GenerateSequence.from(0).to(numberOfTextLines))
-        .apply("Produce text lines", ParDo.of(new DeterministicallyConstructTestTextLineFn()))
-        .apply("Write content to files", TextIO.write().to(filenamePrefix).withOutputFilenames())
-        .getPerDestinationOutputFilenames().apply(Values.<String>create());
+  /** IO IT with no compression. */
+  @RunWith(JUnit4.class)
```

Comment:
This means that this test will be picked up by all test suites (including Java pre-commit), isn't it ? Not sure if we want to do that due to the size of this test. Adding to post-commit tests should be fine.

Reply:
I double-checked that by running the preCommit job on my machine - those are not fired in PreCommit phase. Also, out of curiosity 😄 I investigated a little bit the project's mvn structure:

besides the `@RunWith(JUnit.class)` annotation that is required by JUnit, we have two mvn plugins that look (scan) for tests:
  - surefire (looks for unit tests and searches for classes with *Test suffix)
  - failsafe (looks for integration tests and searches for classes with *IT suffix)

As failsafe is not fired in the PreCommit phase, the tests are not invoked. Please look at [io parent pom](https://github.com/apache/beam/blob/master/sdks/java/io/pom.xml#L77), where failsafe plugin is activated only when io-it profile is active.